airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -304,7 +304,15 @@ def multi_stream_iter(self, log_group, streams, positions=None):
         positions = positions or {s: Position(timestamp=0, skip=0) for s in streams}
         event_iters = [self.log_stream(log_group, s, positions[s].timestamp, positions[s].skip)
                        for s in streams]
-        events = [next(s) if s else None for s in event_iters]
+        events = []
+        for s in event_iters:
+            if not s:
+                events.append(None)
+                continue
+            try:
+                events.append(next(s))
+            except StopIteration:
+                events.append(None)
```

Comment:
This seems.... odd to me. StopIteration inside a `for` loop should already be handled:


```python
def mkiter():
    yield 1
    yield 2
    yield 3
    raise StopIteration()

for a in mkiter():
    print(a)
print("done")
```

When run:

```python
>>> def mkiter():
...     yield 1
...     yield 2
...     yield 3
...     raise StopIteration()
...
>>> mkiter()
<generator object mkiter at 0x10b9d9308>
>>> for a in mkiter():
...     print(a)
...
1
2
3
>>> print("done")
done
```

Reply:
`
>>> # tested in python 3.6.4
... 
>>> def mkiter():
...     yield 1
... 
>>> iter_list = [mkiter(), mkiter()]
>>> value_list = [next(iter) for iter in iter_list]
>>> value_list
[1, 1]
>>> value_list = [next(iter) for iter in iter_list]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
StopIteration
`

I think the StopIteration is thrown because I explicitly call next() here on an iterator which is at the end of iteration. So the for loop is iterating on a list of iterators instead of values in a single iterator.